src/connector/src/parser/postgres.rs

```diff
@@ -106,304 +59,65 @@ fn postgres_cell_to_scalar_impl(
-                    Type::NUMERIC => {
-                        // Currently in order to handle the decimal beyond RustDecimal,
-                        // we use the PgNumeric type to convert the decimal to a string.
-                        // Note: It's only used to map the numeric type in upstream Postgres to RisingWave's varchar.
-                        let res = row.try_get::<_, Option<ScalarAdapter<'_>>>(i);
-                        match res {
-                            Ok(val) => val.and_then(|v| v.into_scalar(DataType::Varchar)),
-                            Err(err) => {
-                                log_error!(name, err, "parse numeric column as pg_numeric failed");
-                                None
-                            }
-                        }
-                    }
-                    _ => {
-                        handle_data_type!(row, i, name, String)
-                    }
-                }
-            }
-        }
-        DataType::Date => {
-            handle_data_type!(row, i, name, NaiveDate, Date)
-        }
-        DataType::Time => {
-            handle_data_type!(row, i, name, chrono::NaiveTime, Time)
-        }
-        DataType::Timestamp => {
-            handle_data_type!(row, i, name, chrono::NaiveDateTime, Timestamp)
-        }
-        DataType::Timestamptz => {
-            handle_data_type!(row, i, name, chrono::DateTime<Utc>, Timestamptz)
+        DataType::Decimal => {
+            // Decimal is more efficient than PgNumeric in ScalarAdapter
+            handle_data_type!(row, i, name, Decimal)
         }
-        DataType::Bytea => {
-            let res = row.try_get::<_, Option<Vec<u8>>>(i);
+        DataType::Varchar | DataType::Int256 => {
+            let res = row.try_get::<_, Option<ScalarAdapter>>(i);
             match res {
-                Ok(val) => val.map(|v| ScalarImpl::from(v.into_boxed_slice())),
+                Ok(val) => val.and_then(|v| v.into_scalar(data_type)),
                 Err(err) => {
                     log_error!(name, err, "parse column failed");
                     None
                 }
             }
         }
-        DataType::Jsonb => {
-            handle_data_type!(row, i, name, serde_json::Value, JsonbVal)
-        }
-        DataType::Interval => {
-            handle_data_type!(row, i, name, Interval)
-        }
-        DataType::List(dtype) => {
-            let mut builder = dtype.create_array_builder(0);
-            // enum list needs to be handled separately
-            if let Kind::Array(item_type) = row.columns()[i].type_().kind()
-                && let Kind::Enum(_) = item_type.kind()
-            {
-                // Issue #1, we use ScalarAdapter instead of Option<ScalarAdapter>
-                let res = row.try_get::<_, Option<Vec<ScalarAdapter<'_>>>>(i);
+        DataType::List(dtype) => match **dtype {
+            // TODO(Kexiang): allow DataType::List(_)
```

Comment:
The priority to support nested list depends on use case.

Reply:
Agree, I will put it on the back burner for now.